vsock-proxy/parent/src/parent.rs

```diff
@@ -186,30 +187,51 @@ async fn send_enclave_exit(enclave_port: &mut AsyncVsockStream) -> Result<(), St
 
-    // Filter the environment variables sent from the parent. The ORIG_ENV_LIST_PATH file contains
-    // the list of variables which were set in the parent at conversion time. Rules for filtering:
-    //  - Always pass HOSTNAME
-    //  - Keep all environment variables which are new (i.e. not present in the parent at conversion
-    //    time)
-    //  - Keep environment variables which were present in the parent container but their values
-    //    have now been updated. The exception to this rule is for the PATH variable.
+    // "_" is a special var setup by bash that points to the currently executed binary
+    // to not leak the binary path into the enclave we remove this var explicitly.
+    runtime_vars.remove("_");
+
+    // The ORIG_ENV_LIST_PATH file contains the list of variables which were set in the parent at conversion time.
     let file =
         File::open(orig_env_path.as_path()).map_err(|e| format!("Unable to find parent's original env variables : {}", e))?;
     let reader = BufReader::new(file);
     for line in reader.lines() {
         let env_line = line.map_err(|e| format!("Unable to read line from file {:?} : {:?}", orig_env_path, e))?;
         // Ill formed env variables will be ignored
-        let env_key_val = env_line.split_once("=").unwrap_or(("", ""));
-        if env_key_val.0 != "HOSTNAME" {
-            info!("Testing if {} does not exist or has been updated.", env_line);
-            runtime_vars.retain(|o| o.0 != env_key_val.0 || (o.0 != "PATH" && (o.0 == env_key_val.0 && o.1 != env_key_val.1)));
+        let parent_env_var = env_line.split_once("=").unwrap_or(("", ""));
+
+        filter_parent_env_from_runtime_envs(&mut runtime_vars, parent_env_var);
+    }
+
+    Ok(runtime_vars.into_iter().collect())
+}
+
+/// Filter the environment variables sent from the parent at runtime.
+/// This function modifies a mutable `HashMap` of runtime environment variables by removing a specified key-value pair if certain conditions are met.
+/// # Rules for filtering are:
+///  - Always pass HOSTNAME
+///  - Keep all environment variables which are new (i.e. not present in the parent at conversion
+///    time)
+///  - Keep environment variables which were present in the parent container but their values
+///    have now been updated. The exception to this rule is for the PATH variable.
+fn filter_parent_env_from_runtime_envs(runtime_env_vars: &mut HashMap<String, String>, parent_env: (&str, &str)) -> () {
+    let (conv_time_env_key, conv_time_env_val) = parent_env;
+
+    if conv_time_env_key != "HOSTNAME" {
+        info!("Testing if {:?} does not exist or has been updated.", parent_env);
+
+        match runtime_env_vars.get(conv_time_env_key) {
+            Some(value) if value == conv_time_env_val || conv_time_env_key == "PATH" => {
+                runtime_env_vars.remove(conv_time_env_key);
+            }
+            _ => {
```

Comment:
Which use case does this block cover? Can we have a unit test for this?

Reply:
Can you be more specific? GIthub doesn't mark the block precisely so I don't understand what block you are curious about, also the whole logic is moved, not redesigned from the previous solution with a `Vec`. If you are asking about the test case with a `PATH` it's: `test_runtime_path_key_is_removed` and for the same values it's tests case: `test_runtime_key_not_removed_when_values_dont_match`.

Reply:
I was referring to the catch all block. I see the `test_runtime_key_not_removed_when_values_dont_match` unit test now.